net/ip.c

```diff
@@ -0,0 +1,343 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+
+#include "lib/assert.h"
+#include "lib/errno.h"
+#include "net/net_internal.h"
+#include <arpa/inet.h>          /* inet_pton, htons */
+#include <netdb.h>              /* hostent */
+#include <stdlib.h>             /* atoi */
+
+#define M0_TRACE_SUBSYSTEM M0_TRACE_SUBSYS_NET
+#include "lib/trace.h"
+
+static const char *ip_family[M0_NET_IP_AF_MAX]      = { "inet",
+							"inet6",
+							"unix" };
+
+static const char *ip_protocol[M0_NET_IP_PROTO_MAX] = { "tcp",
+							"verbs",
+							"stream",
+							"dgram",
+							"o2ib",
+							"lo" };
+
+/**
+ * This function convert the hostname/FQDN format to ip format.
+ * Here hostname can be FQDN or local machine hostname.
+ */
+static int m0_net_hostname_to_ip(char *hostname , char* ip,
+				 struct m0_net_ip_addr *ip_addr)
+{
+	struct hostent  *hname;
+	struct in_addr **addr;
+	int              i;
+	int              n;
+	char            *cp;
+	char             name[M0_NET_IP_STRLEN_MAX];
+
+	M0_ENTRY("Hostname = %s", (char*)hostname);
+	cp = strchr(hostname, '@');
+	if (cp == NULL)
+		return M0_ERR(-EINVAL);
+
+	n = cp - hostname;
+	memcpy(name, hostname, n);
+	name[n] = '\0';
+	if ((hname = gethostbyname(name)) == NULL) {
+		/** get the host info */
```

Comment:
Replace '/** get the host info */'
'/* Get the host info. */'

Reply:
Done